### ISSUE TITLE
Text UI doc picker to /dev/tty ?

### ISSUE THREAD

**Issue by jessebett** [papis/papis#236]
I'm trying to integrate papis into my vim workflow in a more customized way than https://github.com/papis/papis-vim is currently supporting.

One of the issues that I'm having is that I want to call a papis command from inside vim, enter the TUI to pick a doc, and then read the standard output from papis back into vim. However, doing this naively results in reading the entire TUI directly into vim. Is this because the TUI is being sent to standard out? (not sure if that makes sense, even)

I detailed this question [in a stackexchange post](https://stackoverflow.com/questions/58735011/how-to-call-external-interactive-tui-command-interact-and-read-std-output?noredirect=1#comment103781266_58735011) which had the suggestion to have `papis` send the TUI to `/dev/tty` instead of to standard output, and only the returned value gets put into standard out. Supposedly this will make my desired behaviour Just Work.

**Comment by alejandrogallo** [comment]
Yes agreed, I also thought about this the only problem is that it's not clear to me how to be able to do this, specially in a manner that works also on windows. I'm open to suggestions whenever the fix is working everywhere.
A workaround is doing something like this

```vim
function! PapisBibtex(query)
  let l:temp = tempname()
  exec "!papis export -o ".l:temp." --format bibtex ".a:query
  exec ':.read '.l:temp
endfunction
command! -nargs=1 PapisBibtex | call PapisBibtex(<args>)
```

**Comment by jessebett** [comment]
@alejandrogallo right not sure how things work on Windows. 

Is the above`PapisBibtex` for `papis-vim`? I can take a look. In general solving this could resolve a lot of issues with that package, e.g. requiring that you have a certain papis backend. If we could just call out to usual `papis open` and get the document explorer wouldn't need to use vim FZF on a `woosh` backend. And `:PapisView` would just use this. 

Also it could allow for more customization than what's supported there. I opened an issue there(https://github.com/papis/papis-vim/issues/4) about my larger goal that my recent issues have been towards, which is better integration with vimwiki. My papis directory is inside my vimwiki directory, so I'm working toward having a `:PapisLinkToNotes` type function which inserts the markdown syntax something like `[{doc[title]}. {doc[author_list][0]} et al. ({doc[year]})](relative/path/to/notes)` which would allow me to get to the notes file using vimwiki markdown movements. The purpose of #232 for this goal is to also have something like `:PapisViewAndNotes` open the pdf and edit the notes file. I've not determined how this should work yet with my vimwiki ambitions yet. Possibly a function like `PapisViewFromNotes` which is callable from within the notes file and opens the document. Hmm.

Anyway, these are what these recent issues, esp. related to vim integration, are towards...

**Comment by supersambo** [comment]
Unfortunately, I do not know about a solution for windows. But on Linux systems explicitely declaring an output in the [Picker class](https://github.com/papis/papis/blob/68071b072b5453a698f45e7ff73ae86ac5f49156/papis/tui/app.py#L336) such as this might be a solution:

``` python
from prompt_toolkit.output import create_output
import io, os
#Line 336 
output=create_output(stdout=io.TextIOWrapper(io.FileIO(os.open("/dev/tty", os.O_NOCTTY | os.O_RDWR), "r+"))),
```

At least that seems to work for me (don't ask me about the `os.open` arguments :) ). This is just a note for anyone who wants to investigate on that issue...
